certs/trustpin.go

```diff
@@ -0,0 +1,89 @@
+package certs
+
+import (
+	"crypto/x509"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary"
+	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/tuf/utils"
+	"strings"
+)
+
+// TrustPinChecker handles logic for trust pinning for a gun, given a TrustPinConfig
+type TrustPinChecker struct {
+	mode   TrustPinMode
+	gun    string
+	config notary.TrustPinConfig
+}
+
+// TrustPinMode is a type to distinguish possible trust pinning modes
+type TrustPinMode string
+
+const (
+	tofus TrustPinMode = "tofu"
+	ca    TrustPinMode = "ca"
+	certs TrustPinMode = "certs"
+)
+
+// NewTrustPinChecker returns a new TrustPinChecker from a TrustPinConfig for a GUN
+func NewTrustPinChecker(trustPinConfig notary.TrustPinConfig, gun string) (TrustPinChecker, error) {
+	trustPinChecker := TrustPinChecker{config: trustPinConfig, gun: gun}
+	// Determine the mode, and if it's even valid
+	if trustPinConfig.TOFU {
```

Comment:
good catch, I'll change the order!  Good idea on the tests too, I'll add some precedence tests :)